### PR TITLE
NF: cleanup of NoteEditorTest

### PR DESCRIPTION
**Pull request by Arthur-Milchior**

### CODE REVIEW

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTest.kt [hunk 1]
```diff
@@ -72,26 +77,9 @@ abstract class NoteEditorTest protected constructor() {
              java.lang.AssertionError: Activity never becomes requested state "[DESTROYED]" (last lifecycle transition = "PAUSED")
              at androidx.test.core.app.ActivityScenario.waitForActivityToBecomeAnyOf(ActivityScenario.java:301)
               */
-            val invalid = Build.VERSION_CODES.N_MR1
-            val integers = ArrayList(listOf(invalid))
-            integers.addAll(invalidSdks!!)
-            return integers
+            return mutableListOf(Build.VERSION_CODES.N_MR1).apply { addAll(invalidSdks) }
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTest.kt:80]: Delete the kotlin cleanup on this property as it was fixed.

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTest.kt [hunk 2]
```diff
@@ -72,26 +77,9 @@ abstract class NoteEditorTest protected constructor() {
...
         }
-    protected open val invalidSdks: List<Int>?
-        get() = ArrayList()
+    protected open val invalidSdks = listOf<Int>()
     protected val targetContext: Context
         get() = InstrumentationRegistry.getInstrumentation().targetContext
-
-    init {
-        @KotlinCleanup("change to variable init")
-        // Rules mean that we get a failure on API 25.
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTest.kt:87]: We should keep these comments on the class.

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.kt [hunk 3]
```diff
@@ -33,12 +34,12 @@ import java.util.concurrent.atomic.AtomicReference
...
 @RunWith(AndroidJUnit4::class)
 class NoteEditorTabOrderTest : NoteEditorTest() {
-    override val invalidSdks: List<Int>
+    override val invalidSdks =
         /*
             java.lang.AssertionError:
             Expected: is "a"
          */
-        get() = listOf(30)
+        listOf(Build.VERSION_CODES.N_MR1)
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.kt:42]: Build.VERSION_CODES.N_MR1 integer value is 25 not 30 like in the original code.

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.kt [hunk 4]
```diff
@@ -87,8 +88,8 @@ class NoteEditorTabOrderTest : NoteEditorTest() {
                 wrapped.set(t)
             }
         }
-        if (wrapped.get() != null) {
-            throw wrapped.get()!!
+        wrapped.get()?.let {
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.kt:91]: Cleanup was done so delete the cleanup annotation reference on the class.

**Review by lukstbit — CHANGES_REQUESTED** [review]
Note that with the changes in NoteEditorTest one of its subclasses(FieldEditLineTest) crashes due to a null pointer exception.

The cause is isInvalid accessing invalidSdkImpl which accesses invalidSdk at the wrong time.

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.kt (hunk 3 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.kt]: I'm confused. I can't imagine how I got this wrong

**Review by Arthur-Milchior — COMMENTED** [review]
Thanks @lukstbit
Sorry I missed your review. Corrected

**Pull request closed without merging**